src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    // Analyzer and code fix tests are both in this file
+    // "Do not separate analyzer tests from code fix tests"
+    // https://github.com/dotnet/roslyn-analyzers/blob/main/docs/NetCore_GettingStarted.md#definition-of-done
+    public class AvoidConstArraysTests
+    {
+        #region C# Tests
+
+        [Fact]
+        public async Task CA1839_CSharp_IdentifyConstArrays()
+        {
+            // Implicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine(new[]{ 1, 2, 3 });
+    }
+}
+", VerifyCS.Diagnostic().WithSpan(8, 27, 8, 43), @"
+using System;
+
+public class A
+{
+    private static readonly int[] value = new[] { 1, 2, 3 };
+
+    public void B()
+    {
+        Console.WriteLine(value);
+    }
+}
+");
+
+            // Explicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine(new int[]{ 1, 2, 3 });
+    }
+}
+", VerifyCS.Diagnostic().WithSpan(8, 27, 8, 47), @"
+using System;
+
+public class A
+{
+    private static readonly int[] value = new int[] { 1, 2, 3 };
+
+    public void B()
+    {
+        Console.WriteLine(value);
+    }
+}
+");
+
+            // Nested arguments
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine(string.Join("" "", new[] { ""Cake"", ""is"", ""good"" }));
+    }
+}
+", VerifyCS.Diagnostic().WithSpan(8, 44, 8, 74), @"
+using System;
+
+public class A
+{
+    private static readonly string[] value = new[] { ""Cake"", ""is"", ""good"" };
```

Comment:
It may be difficult to adhere to user preferences. In this case, I'd probably stick with prefixing static fields with `s_`. Tagging @sharwell for opinion.